### PR TITLE
Sms allowance dashboardbug

### PR DESCRIPTION
**Pull request by A-Shumway42**
## Description

- Free sms allowance wasn't updating on the dashboard. Although the value would change when switching a service from trial mode to live, it still wasn't updating in other places. The dashboard call was passing in a year value that would always bring back `40000` free sms allowance. 
- This uncovered another issue where the dashboard value wouldn't increment due to values returning incorrectly from `get_current_financial_year()`. This method will now return the `current_year` if the month is before October, else it will return the next year. The dashboard is now incrementing properly.

Screen Shot 2024-02-07 at 9 55 25 AM

- A service in trial mode will now increment the free sms allowance count.

Screen Shot 2024-02-07 at 2 42 26 PM

- The usage page broke after these time changes. The usage page will now display like so and move to a more calendar year approach, displaying all months and their sms values sent during that month.

Screen Shot 2024-02-07 at 9 36 24 AM

## TODO

- [  ]  Do we want to keep the `fiscal year` text on each tab or change it to `calendar year`?

## Security Considerations
None

### CODE REVIEW

File: app/utils/time.py [hunk 1]
```diff
@@ -11,7 +11,7 @@ def get_current_financial_year():
     now = datetime.now(preferred_tz)
     current_month = int(now.strftime("%-m"))
     current_year = int(now.strftime("%Y"))
-    return current_year if current_month > 9 else current_year - 1
+    return current_year if current_month < 10 else current_year + 1
```
Comment (ccostino) [inline app/utils/time.py:14]: Just checking my understanding here:  this is the switch to basing the fiscal year starting in October, so if we're before October we're in the current calendar year, and if we're after then technically we're in the next calendar year, right?

File: tests/app/main/views/organizations/test_organizations.py [hunk 2]
```diff
@@ -388,7 +388,7 @@ def test_organization_services_shows_live_services_and_usage(
 
     client_request.login(active_user_with_permissions)
     page = client_request.get(".organization_dashboard", org_id=ORGANISATION_ID)
-    mock.assert_called_once_with(ORGANISATION_ID, 2019)
+    mock.assert_called_once_with(ORGANISATION_ID, 2020)
```
Comment (ccostino) [inline tests/app/main/views/organizations/test_organizations.py:391]: Ha!  We missed an `organisation` to `organization change here it looks like... 😆 

(No need to address here, just calling it out!)

File: tests/app/main/views/test_dashboard.py [hunk 3]
```diff
@@ -683,17 +683,29 @@ def test_should_show_monthly_breakdown_of_template_usage(
...
+            "February",
+            "March",
+            "April",
+            "May",
+            "June",
+            "July",
+            "August",
+            "September",
+            "October",
+            "November",
```
Comment (ccostino) [inline tests/app/main/views/test_dashboard.py:707]: Do we need December in here, too?

```suggestion
            "November",
            "December",
```

File: tests/app/main/views/test_dashboard.py [hunk 4]
```diff
@@ -666,12 +666,12 @@ def test_should_show_redirect_from_template_history(
     )
 
 
-@freeze_time("2017-01-01 12:00")  # 4 months into 2016 financial year
+@freeze_time("2017-1-1 12:00")  # Switching to calendar year
```
Comment (ccostino) [inline tests/app/main/views/test_dashboard.py:669]: For the most part all of this work is based on fiscal year still, yes?  I see the date itself isn't actually different so it doesn't look like there's a real change here, but the comment caught my eye if the intent was to actually switch to calendar year here.

In this case I'd also recommend keeping the leading zeros as that is generally expected when dealing with date and datetime values in Python for days and months (e.g., like in [`freezegun`'s own docs](https://github.com/spulec/freezegun?tab=readme-ov-file#decorator)):

```suggestion
@freeze_time("2017-01-01 12:00")  # Switching to calendar year
```

**Review by ccostino — COMMENTED** [review]
Thanks, @A-Shumway42, I know there was a lot of back and forth with this one!

I had a few questions as I went through everything, and the big on is whether this should be focused purely on fiscal year vs. switching entirely to calendar year.  It looks like this is doing a little of both and I'm concerned that this might introduce more confusion.

Tagging in @stvnrlly here for a gut check as well, to make sure my thinking is right:  for the purposes of this PR and the fact that the code was dealing with the concept of a fiscal year previously, should we keep these changes to be centered around our fiscal year (October - September) for now and switch to calendar year as a follow-on later on?

Or, do we feel we have a clear enough picture here that we could just finish flipping everything over to calendar year entirely and be done with it?

It looks like we could make the full switch to calendar year, and I'm leaning toward that just to get us clear and done with it.

However, if I'm missing anything on that front or there's still some open questions about it, then I'd say we focus the changes on our fiscal year for now and follow-up with a separate issue and PR for the calendar year switch.

File: app/utils/time.py (hunk 1 above)
Comment (A-Shumway42) [reply app/utils/time.py:14]: Yeah, exactly!

File: tests/app/main/views/test_dashboard.py (hunk 3 above)
Comment (A-Shumway42) [reply tests/app/main/views/test_dashboard.py:707]: I believe so. Let me run the tests and see what I potentially missed here.

File: tests/app/main/views/test_dashboard.py (hunk 4 above)
Comment (A-Shumway42) [reply tests/app/main/views/test_dashboard.py:669]: Yeah, we are still based on fiscal year from October to September. I probably got ahead of myself on this comment. Thank you for the explanation!

File: app/utils/time.py (hunk 1 above)
Comment (ccostino) [reply app/utils/time.py:14]: Thanks!

File: tests/app/main/views/test_dashboard.py (hunk 4 above)
Comment (ccostino) [reply tests/app/main/views/test_dashboard.py:669]: No worries, and you're welcome!

**Comment by ccostino** [comment]
@A-Shumway42 and @stvnrlly, circling back here on this one, and I think aside from the merge conflict this should now be good to go?

**Review by ccostino — APPROVED** [review]
Awesome, thanks @A-Shumway42!

**Pull request merged**